### PR TITLE
add txServiceUrl in safe config

### PR DESCRIPTION
**Pull request by RnkSngh**
PR to add support for directly integrating wiht tx service urls that aren't the default operated by safe. 

this is required for adding multisig to bob & mode 

## Release Notes

- **New Features**
  - Introduced an optional `txServiceUrl` property in multisig configuration schemas for enhanced flexibility.
  
- **Improvements**
  - Streamlined function signatures for `proposeTransaction` and `executeMultisigTx`, consolidating multisig configuration into a single object.
  - Updated transaction execution process to handle multisig account information more efficiently.

- **Bug Fixes**
  - Maintained consistent error handling in the transaction sending process, ensuring robust performance.

### CODE REVIEW

File: src/evm/schemas/multisig.ts [hunk 1]
```diff
@@ -37,6 +38,7 @@ export const initializedMultisig = z.object({
   privateKey: z.string().min(1),
   safeAddress: z.string().min(1),
   wallet: wallet,
+  txServiceUrl: z.string().optional(),
```
[the remark this answers is not in the document]
Comment (RnkSngh) [reply src/evm/schemas/multisig.ts:41]: should be handled by zod.safeParseSchema

File: src/multisig/safe.ts [hunk 2]
```diff
@@ -37,36 +39,34 @@ export const newSafeFromOwner = async (
...
-  const apiKit = new SafeApiKit({ chainId });
+
+  const apiKit = new SafeApiKit({ chainId:BigInt(initializedMultisig.chainId), txServiceUrl: initializedMultisig.txServiceUrl});
 
   const txProposer = await Safe.init({
     provider: rpcUrl,
-    signer: proposerPrivateKey,
-    safeAddress: safeAddress,
+    signer: initializedMultisig.wallet.privateKey,
+    safeAddress: initializedMultisig.safeAddress,
```
[the remark this answers is not in the document]
Comment (RnkSngh) [reply src/multisig/safe.ts:56]: accessing the wallet property ensures we access the correctly parsed  (with env substitution) private key  vs a raw entry in account config

**Review by dshiell — APPROVED** [review]
are txServiceUrls "trustworthy"?

**Comment by RnkSngh** [comment]
> are txServiceUrls "trustworthy"?

I think they are, because they tend to be operated by chains themselves (e.g. the urls are superchain or gobob.xyz) 

But also, we don't _need_ a lot of trust assumptions here. 

We only use this api service to sign and persist transactions which we can assume are public. So since there aren't any concerns with leaking the info we send them (in fact, the last step _is_ broadcasting the signed tx publicly,  the only way they could grief us is through losing our txs. But that seems unlikely

**Pull request merged**